### ISSUE TITLE
Migration - grunt work

### ISSUE THREAD

**Issue by PaulieScanlon** [PaulieScanlon/mdx-embed#7]
The current plugin `gatsby-mdx-embed` uses Storybook v5 `.stories.mdx` The new project `mdx-embed` uses the laster version of Storybook v6

There are some difference between how components and prop tables are defined between the two versions.

eg v5 Storybook CodePen stories.mdx looks like this

```javascript
import { Meta, Preview, Props } from '@storybook/addon-docs/blocks'

import { CodePen } from '../../../@pauliescanlon/gatsby-mdx-embed/src/components/CodePen'

<Meta title="Components/CodePen" component={CodePen} />

# CodePen

Display a CodePen by including the component in your `.mdx`. The `codePenId` prop is required.

<Props of={CodePen} />

## Basic usage

<Preview>
  <CodePen codePenId="PNaGbb" />
</Preview>

## Tabs

<Preview>
  <CodePen codePenId="PNaGbb" tabs={['js', 'result']} />
</Preview>

```

BUT v6 Storybook CodePen stories.mdx looks like this

```javascript
import { Meta, Story, Source, ArgsTable } from '@storybook/addon-docs/blocks';

import { CodePen } from '../../../packages/mdx-embed/src/components/CodePen';

<Meta title="Components/CodePen" component={CodePen} />

# CodePen

Display a CodePen by including the component in your `.mdx`. The `codePenId` prop is required.

<ArgsTable of={CodePen} />

## Usage

<Story name="usage">
  <CodePen codePenId="PNaGbb" />
</Story>

<Source ID="usage" />

## Tabs

<Story name="tabs">
  <CodePen codePenId="PNaGbb" tabs={['js', 'result']} />
</Story>

<Source ID="tabs" />

## Help

<Story name="help" />

To embed a CodePen using the `<CodePen />` component grab the `codePenId` from the URL.

https://codepen.io/team/codepen/pen/PNaGbb

The `codePenId` is this bit.

PNaGbb

Then use it in the component like this.

<CodePen codePenId="PNaGbb" />

```

A few things to note are the differences between some of the Storybook components. `Props` has been deprecated and `ArgsTable` is the new way to read TypeScript interface details and display them in the table you see at the top of the story. 

The way you defined a `Story` and a `Source` is also different in v6

Also a note on the component import. We'll have to use absolute path to the component because If we import the component like this `import { CodePen } from "mdx-embed"` we don't get the props... which is annoying!

All components and their respective stories need to be migrated from the old package to the new package and all those annoying changes to the Storybook components needs to be done.

Old component location: 
`@pauliescanlon/gatsby-mdx-embed/src/components`

New component location:
`packages/mdx-embed/src/components`

Old docs location:
`docs/src/components`

New docs location:
`docs/src/components`

There's one additional thing... in the old repo there's a separate .mdx in  `docs/src/pages/3.help.stories.mdx`

This isn't the best idea, but it was a workaround to a Storybook v5 mdx nested story/navigation label problem... that's now fixed so i think we should move each of the help instructions to the actual stories file for each of the components. I've done the CodePen one in the new repo already.
(CodePen will actually need to get updated at some point as there's an open PR on the old repo - i'll take care of this at some point)

My suggestion for the migration if you'd like to help out is do one component to see get familiar with it. YouTube should be an easy one, create a branch / PR and let's see how much of a ball ache this is gonna be. 

I'm not expecting you to do all of them, perhaps i'll take the Twitter, Flickr and Instagram ones as they have some extra script injection stuff which i need to double check will still work outside of the Gatsby plugin... how's that sound?

## Component List

- [x] Buzzprout
- [x] CodePen
- [x] CodeSandbox
- [x] Egghead
- [x] Flickr
- [x] Gist
- [x] Instagram
- [x] Lbry
- [x] Pinterest
- [x] Simplecast
- [x] SoundCloud
- [x] Spotify
- [x] Strava
- [x] TikTok
- [x] Twitch
- [x] Twitter
- [x] Vimeo
- [x] Wikipedia
- [x] YouTube

**Comment by spences10** [comment]
Sounds super! 

So you want a PR for each component?

I was thinking about doing it all in one branch, but guessing your approach will be a good test my tolerance! 😂

I'll make a feature branch to add YouTube, cheers buddy!

**Comment by spences10** [comment]
Closing 😅

**Comment by spences10** [comment]
Forgot about #27 🤦‍♀️

**Comment by spences10** [comment]
This closed with #27 now @PaulieScanlon?

**Comment by PaulieScanlon** [comment]
@spences10 yeah, Flickr was the last one to go it. This can now be closed!
